### ISSUE TITLE
Clarify dependency on NumPy (related to maskedtensor?)

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#76656]
### 🐛 Describe the bug

I installed torch cpu version on a new python3.8 ubuntu 20.04 system via pip. From what I understood, it did not install NumPy as dependency.

When I import torch, I get:
```
/usr/local/lib/python3.8/dist-packages/torch/_masked/__init__.py:223: UserWarning: Failed to initialize NumPy: numpy.core.multiarray failed to import (Triggered internally at  ../torch/csrc/utils/tensor_numpy.cpp:68.)
  example_input = torch.tensor([[-3, -2, -1], [0, 1, 2]])
```

This has a lot of confusing details. If NumPy is needed, it should be specified as pip dependency and installed automatically.

If it's not strictly required, I propose that either no warning is shown on import or the warning is more clear: remove the file paths, line number, example tensor and clearly explain why NumPy is needed + point to URL that explains all the details clearly.

I think that it's not strictly needed and that maybe something about this was forgotten during maskedtensor integration.

On testing note: there should be a test of PyTorch install and import on systems without numpy to check that all works 

cc @malfet

### Versions

1.11.0+cpu

**Comment by gchanan** [comment]
@cpuhrsch

**Comment by cpuhrsch** [comment]
@vadimkantorov - thanks for filing this issue. Does this reproduce on a recent nightly for you? This was likely fixed alongside some changes to the documentation for masked operations. cc @pearu.

**Comment by malfet** [comment]
@cpuhrsch this should reproduce for anyone if PyTorch is installed, but not NumPy. And yes, we probably should call a slightly better constructor in that case.
